File: Test_suite/Java/factorial.java
import java.math.BigInteger;

public class FactorialCalculator {

    public static BigInteger factorialIterative(int n) {
        BigInteger result = BigInteger.ONE;
        for (int i = 2; i <= n; i++) {
            result = result.multiply(BigInteger.valueOf(i));
        }
        return result;
    }

    public static void main(String[] args) {
        int number = 100000; // You can adjust this number

        // Memory usage estimation
        long beforeUsedMem = Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory();
        long startTime = System.currentTimeMillis();
        BigInteger result = factorialIterative(number);
        long endTime = System.currentTimeMillis();
        // Execute your code (already done above)
        long afterUsedMem = Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory();
        long actualMemUsed = afterUsedMem - beforeUsedMem;
        long actualMemUsedMB = actualMemUsed / ( 1024 * 1024 );

        System.out.println("The factorial of " + number + " is calculated.");
        System.out.println("Execution Time: " + (endTime - startTime) + " milliseconds");

        System.out.println("Estimated Memory Usage: " + actualMemUsedMB + " bytes");
    }
}

File: Test_suite/Python/factorial.py
import time
import resource

def factorial_iterative(n):
    result = 1
    for i in range(2, n + 1):
        result *= i
    return result

def get_memory_usage():
    # Get the current memory usage in kilobytes
    memory_kb = resource.getrusage(resource.RUSAGE_SELF).ru_maxrss
    # Convert kilobytes to megabytes
    memory_mb = memory_kb / 1024.0
    return memory_mb

def main():
    number = 100000  # You can adjust this number
    start_time = time.time()
    result = factorial_iterative(number)
    end_time = time.time()

    print(f"The factorial of {number} is calculated.")
    print(f"Execution Time: {end_time - start_time} seconds")

if __name__ == "__main__":
    before_memory = get_memory_usage()
    main()
    after_memory = get_memory_usage()

    # Calculate peak memory usage during the execution
    peak_memory = max(before_memory, after_memory)
    print(f"Peak Memory Usage: {peak_memory} MiB")
